### PR TITLE
ci: add github workflow

### PR DESCRIPTION
**Pull request by TheMythologist**
Related to #59 

Unsure if you want to use black or autopep8 (or both), this workflow runs black.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      - name: Format with black
+        run: poetry run black .
+      - name: Commit changes if any
+        run: |
+          git config user.name "GitHub Actions"
+          git config user.email "action@github.com"
+          if output=$(git status --porcelain) && [ ! -z "$output" ]; then
+            git commit -m "style: Automatic code formatting" -a
+            git push
+          fi
```
Comment (brylie) [inline .github/workflows/test.yml:37]: It would feel a bit safer if CI didn't automatically commit changes to the code, as unexpected things might happen. So, I'd prefer someone to verify the errors and manually fix them.

Would it be possible to warn the developer rather than commit the changes?

**Review by brylie — CHANGES_REQUESTED** [review]

**Comment by brylie** [comment]
@TheMythologist, thank you for this contribution :-)

I've added a small request about not auto-committing changes. I'm still a bit nervous about GitHub Actions being able to commit code to a repo without human review.

File: .github/workflows/test.yml (hunk 1 above)
Comment (TheMythologist) [reply .github/workflows/test.yml]: Changed by https://github.com/GeriLife/companionship-care/pull/61/commits/38e8facd158db857ddcbe19b4601cc84a7773fef

**Comment by brylie** [comment]
> Unsure if you want to use black or autopep8 (or both), this workflow runs black.

What would be the best strategy, Black alone or Black + autopep8? What would be some added benefits with Black + autopep8?

**Comment by TheMythologist** [comment]
A good explanation would be this: https://blog.frank-mich.com/python-code-formatters-comparison-black-autopep8-and-yapf/

_Note_: The site is slightly outdated, black is now configurable with many options.

**TLDR**: Black formats the code more than pep-8 standards, whereas autopep8 only formats the code to follow pep-8 code styles. So formatting with autopep8 is redundant if black is already used to format code.

**Comment by TheMythologist** [comment]
Also apologies for the last 2 commits, I'm referencing [black's documentation](https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html)

**Comment by brylie** [comment]
Ok, this PR is looking good. Is it ready for review, or are you planning other changes?

Relatedly, what would you recommend using to measure and report on test coverage for this project?

**Comment by TheMythologist** [comment]
Yup the PR is ready :)

I would recommend coverage - it's can be easily integrated with most python testing tools such as pytest (which I personally use). Here's their documentation: https://coverage.readthedocs.io/en/6.4.4/

**Comment by brylie** [comment]
> I would recommend coverage

Yes, that is also what Code Climate recommends. Do you think we could include adding Coverage in this pull request, as part of a comprehensive CI process? If so, I can configure the test reporter ID as an environment variable.

https://docs.codeclimate.com/docs/configuring-test-coverage

**Comment by TheMythologist** [comment]
Sorry but I'm not familiar enough with coverage to include it in this PR 😅

**Comment by brylie** [comment]
> Sorry but I'm not familiar enough with coverage to include it in this PR

Me either. All good :-)

Is this PR ready for release now?

**Comment by TheMythologist** [comment]
Yup it is

**Comment by brylie** [comment]
Oh no! I tried to resolve the conflicts, and it created a commit that made this branch out of sync with `main`. I've been rebasing into main, so it is necessary to rebase into other development branches, such as this, to keep things in sync. Could you rebase `main` into your local branch and force-push it to GitHub?
